Tests/ML/augmentations/test_image_transforms.py

```diff
@@ -0,0 +1,97 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import random
+
+import numpy as np
+import pytest
+
+import torch
+
+from scipy.ndimage import gaussian_filter, map_coordinates
+
+from InnerEye.ML.augmentations.image_transforms import AddGaussianNoise, ElasticTransform, ExpandChannels, RandomGamma
+
+
+def test_add_gaussian_noise() -> None:
+    """
+    Tests functionality of add gaussian noise
+    """
+    np.random.seed(1)
+    tensor_img = torch.ones([1, 1, 256, 256], dtype=torch.float)
+    tensor_img[..., 100:150, 100:200] = 1 / 255.
+    # Input tensor [C, Z, H, W]
+    torch.manual_seed(10)
+    transformed = AddGaussianNoise(std=0.05, p_apply=1)(tensor_img)
+    torch.manual_seed(10)
+    noise = torch.randn(size=(1, 256, 256)) * 0.05
+    assert torch.isclose(torch.clamp(tensor_img + noise, tensor_img.min(), tensor_img.max()),  # type: ignore
+                         transformed).all()
+    # Check that it applies the same transform to all slices
+    tensor_img = torch.ones([2, 2, 256, 256], dtype=torch.float)
+    tensor_img[..., 100:150, 100:200] = 1 / 255.
+    # Input tensor [C, Z, H, W]
+    torch.manual_seed(10)
+    transformed = AddGaussianNoise(std=0.05, p_apply=1)(tensor_img)
+    torch.manual_seed(10)
+    noise = torch.randn(size=(1, 256, 256)) * 0.05
+    assert torch.isclose(torch.clamp(tensor_img + noise, tensor_img.min(), tensor_img.max()),  # type: ignore
+                         transformed).all()
+
+
+def test_elastic_transform() -> None:
+    """
+    Tests elastic transform
+    """
+    image = torch.ones([2, 2, 256, 256]) * 255.
+    image[..., 100:150, 100:200] = 1
+
+    # Computed expected transform
+    np.random.seed(7)
+    np.random.random(1)
+
+    shape = (256, 256)
+    dx = gaussian_filter((np.random.random(shape) * 2 - 1), 4, mode="constant", cval=0) * 34
```

Comment:
I'm not sure this really is the best way of writing a unit test for this transform. It effectively means that you write the code twice, but still get no confirmation if the code is doing the right thing.
Alternative suggestion for a test:
- you have a "real image" (showing something that humans can interpret) on disk
- you transform it
- you compare the result to another stored image on disk

I'm normally not a fan of stored files, but here it could help because you could quickly take a look at the transformation result and check if it makes sense.

Also, it could help to have tests for p_apply=0 and 1

Reply:
Updated, used one of the test pngs with already have in the repo